psi4/driver/driver.py

```diff
@@ -105,6 +104,27 @@ def _find_derivative_type(ptype, method_name, user_dertype):
 
     return dertype
 
+
+def _energy_is_invariant(gradient, stationary_criterion=1.e-2):
+    """Polls options and probes `gradient` to return whether current method
+    and system expected to be invariant to translations and rotations of
+    the coordinate system.
+
+    """
+    stationary_point = gradient.rms() < stationary_criterion  # 1.e-2 pulled out of a hat
+
+    efp = core.get_active_efp()
+    efp_present = efp.nfragments() > 0
+
+    translations_projection_sound = (not core.get_option('SCF', 'EXTERN') and
+                                     not core.get_option('SCF', 'PERTURB_H') and
+                                     not efp_present)
+    rotations_projection_sound = (translations_projection_sound and 
+                                  stationary_point)
+
+    return translations_projection_sound, rotations_projection_sound
```

Comment:
Yea, I like this too. Single place to find this kind of information.

Reply:
Nice!  This is exactly what I was thinking when nagging about a centralized routine before.